From cf3c220a8fc239d43aafaad23a527f57e36a88fa Mon Sep 17 00:00:00 2001
From: Larisa Grigore <larisa.grigore@nxp.com>
Date: Wed, 31 Mar 2021 14:56:14 +0300
Subject: [PATCH 15/42] s32g3: dts: Update mc_cgm0 definition

In S32G3 case, GMAC implementation was changed compared to
S32G2. Now all GMAC related clocks derive from MC_CGM6 instead
of MC_CGM0 so remove any reference to GMAC from MC_CGM0.

Upstream-Status: Pending

Issue: ALB-6496
Signed-off-by: Larisa Grigore <larisa.grigore@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/dts/fsl-s32-gen1.dtsi |  2 +-
 arch/arm/dts/fsl-s32g398a.dtsi | 57 ++++++++++++++++++++++++++++++++++
 2 files changed, 58 insertions(+), 1 deletion(-)

diff --git a/arch/arm/dts/fsl-s32-gen1.dtsi b/arch/arm/dts/fsl-s32-gen1.dtsi
index 4a72cfb340..1f39d52274 100644
--- a/arch/arm/dts/fsl-s32-gen1.dtsi
+++ b/arch/arm/dts/fsl-s32-gen1.dtsi
@@ -217,7 +217,7 @@
 				<&clks S32GEN1_CLK_ARM_PLL_DFS1>,
 				<&clks S32GEN1_CLK_ARM_PLL_DFS2>;
 			assigned-clock-rates =
-				<800000000>,
+				<S32GEN1_CLK_ARM_PLL_DFS1_FREQ>,
 				<800000000>;
 		};
 
diff --git a/arch/arm/dts/fsl-s32g398a.dtsi b/arch/arm/dts/fsl-s32g398a.dtsi
index 1c24a00048..a086185d88 100644
--- a/arch/arm/dts/fsl-s32g398a.dtsi
+++ b/arch/arm/dts/fsl-s32g398a.dtsi
@@ -8,6 +8,7 @@
  * (at your option) any later version.
  */
 /dts-v1/;
+
 #include "fsl-s32g.dtsi"
 / {
 	model = "NXP S32G3XX";
@@ -28,5 +29,61 @@
 		device_type = "memory";
 		reg = <0 0x34000000 0 0x1400000>;
 	};
+};
 
+/delete-node/ &mc_cgm0;
+&clks {
+	mc_cgm0: mc_cgm0@40030000 {
+		compatible = "fsl,s32gen1-mc_cgm0";
+		reg = <0x0 0x40030000 0x0 0x3000>;
+
+		assigned-clocks =
+			<&clks S32GEN1_CLK_MC_CGM0_MUX0>,
+			<&clks S32GEN1_CLK_MC_CGM0_MUX3>,
+			<&clks S32GEN1_CLK_MC_CGM0_MUX4>,
+			<&clks S32GEN1_CLK_MC_CGM0_MUX5>,
+			<&clks S32GEN1_CLK_MC_CGM0_MUX7>,
+			<&clks S32GEN1_CLK_MC_CGM0_MUX8>,
+			<&clks S32GEN1_CLK_MC_CGM0_MUX12>,
+			<&clks S32GEN1_CLK_MC_CGM0_MUX14>,
+			<&clks S32GEN1_CLK_MC_CGM0_MUX16>,
+			<&clks S32GEN1_CLK_PER>,
+			<&clks S32GEN1_CLK_XBAR_2X>,
+			<&clks S32GEN1_CLK_FTM0_REF>,
+			<&clks S32GEN1_CLK_FTM1_REF>,
+			<&clks S32GEN1_CLK_CAN_PE>,
+			<&clks S32GEN1_CLK_LIN_BAUD>,
+			<&clks S32GEN1_CLK_SPI>,
+			<&clks S32GEN1_CLK_QSPI_2X>,
+			<&clks S32GEN1_CLK_SDHC>;
+		assigned-clock-parents =
+			<&clks S32GEN1_CLK_ARM_PLL_DFS1>,
+			<&clks S32GEN1_CLK_PERIPH_PLL_PHI1>,
+			<&clks S32GEN1_CLK_PERIPH_PLL_PHI1>,
+			<&clks S32GEN1_CLK_PERIPH_PLL_PHI1>,
+			<&clks S32GEN1_CLK_PERIPH_PLL_PHI2>,
+			<&clks S32GEN1_CLK_PERIPH_PLL_PHI3>,
+			<&clks S32GEN1_CLK_PERIPH_PLL_DFS1>,
+			<&clks S32GEN1_CLK_PERIPH_PLL_DFS3>,
+			<&clks S32GEN1_CLK_PERIPH_PLL_PHI7>;
+		assigned-clock-rates =
+			<0>,
+			<0>,
+			<0>,
+			<0>,
+			<0>,
+			<0>,
+			<0>,
+			<0>,
+			<0>,
+			<80000000>,
+			<S32GEN1_XBAR_2X_FREQ>,
+			<40000000>,
+			<40000000>,
+			<80000000>,
+			<125000000>,
+			<100000000>,
+			<400000000>,
+			<400000000>;
+	};
 };
-- 
2.25.1

